src/reducers/ApiKeyReducer/reducer.js

```diff
@@ -0,0 +1,22 @@
+import * as action_types from './constants';
+
+const initialState = {
+  currentKey: null,
+};
+
+function apiKeyReducer(state = initialState, action) {
+  switch(action.type) {
+    case action_types.SET_API_KEY: {
+      return Object.assign({...state}, {
+        currentKey: action.data,
```

Comment:
Api key should be persisted in session/local storage, as then it's not lost on refresh